apps/src/codebridge/utils/isDuplicateFileName.ts

```diff
@@ -1,41 +1,45 @@
 import {ProjectFile} from '@codebridge/types';
 
-import codebridgeI18n from '@cdo/apps/codebridge/locale';
 import {ProjectFileType} from '@cdo/apps/lab2/types';
 
-type CheckForDuplicateFilenameArgs = {
+type IsDuplicateFileNameArgs = {
   fileName: string;
   folderId: string;
   projectFiles: Record<string, ProjectFile>;
   isStartMode: boolean;
   validationFile?: ProjectFile;
 };
 
-export const checkForDuplicateFilename = ({
+export enum DuplicateFileError {
+  DUPLICATE_FILE = 'DUPLICATE_FILE',
+  DUPLICATE_SUPPORT_FILE = 'DUPLICATE_SUPPORT_FILE',
+}
+
+export const isDuplicateFileName = ({
   fileName,
   folderId,
   projectFiles,
   isStartMode,
   validationFile,
-}: CheckForDuplicateFilenameArgs) => {
-  let message = undefined;
+}: IsDuplicateFileNameArgs) => {
   // The validation file is in the project files in start mode.
   if (!isStartMode && validationFile?.name === fileName) {
-    message = codebridgeI18n.duplicateSupportFileError({fileName});
+    return DuplicateFileError.DUPLICATE_SUPPORT_FILE;
   } else {
     const existingFile = Object.values(projectFiles).find(
       f => f.name === fileName && f.folderId === folderId
     );
     if (existingFile) {
-      message = codebridgeI18n.duplicateFileError({fileName});
       if (
         existingFile.type === ProjectFileType.SUPPORT ||
         existingFile.type === ProjectFileType.VALIDATION
       ) {
-        message = codebridgeI18n.duplicateSupportFileError({fileName});
+        return DuplicateFileError.DUPLICATE_SUPPORT_FILE;
+      } else {
+        return DuplicateFileError.DUPLICATE_FILE;
       }
     }
   }
 
-  return message;
+  return false;
```

Comment:
This function is super complicated, and if we could break it apart it'd be great. I'm not completely positive I follow the logic well enough to do it though.

So I opted to do it so it returns a truthy value or false, as opposed to a straight boolean. Insert your own jokes about ternary booleans here. 😝

Reply:
Can you add a comment explaining that this either returns an enum value or false?